### DISCUSSION TITLE
How do I change the performance settings in nvim-cmp

### DISCUSSION THREAD

**Discussion by ngaloppo** [LazyVim/LazyVim#2187] [General]
I'm trying to change the `cmp.performance.debounce` configuration value during setup. I already have a custom config to set up supertab behavior (as per LazyVim docs). How do I adapt it? 

```lua
  {
    "hrsh7th/nvim-cmp",
    ---@param opts cmp.ConfigSchema
    opts = function(_, opts)
    --- ...
    end,
}
```

**Comment by dpetka2001** [comment]
You could add `opts.performance.debounce = ....` I believe inside the `opts` function.

**Reply by ngaloppo** [reply]
That doesn't work. opts doesn't have a performance field. I think the way these config value functions works could be improved. What is this `opts` table, really? And how do you extend existing nested  table(s)? I also tried `opts.performance = { debounce = 1000 }` which is syntactically correct but I don't think that it gets merged into the setup function, I.e. it doesn't properly set the `cmp` config value.

**Reply by ngaloppo** [reply]
I also tried the following, but that complains that `opts.performance` doesn't exist:

```lua
      opts.performance = vim.tbl_extend("force", opts.performance, { debounce = 1000 })
```

**Reply by dpetka2001** [reply]
```lua
  {
    "hrsh7th/nvim-cmp",
    opts = {
      performance = {
        debounce = 1000,
      },
    },
  },
```
Check if this works out.

**Reply by ngaloppo** [reply]
Would this work as an additional section? ie. have two sections for nvim-cpp: one for the supertab behavior, and another for the performance setting. I just tried that, I'm not sure it took effect. Is there a way to inspect the cmp settings?

**Reply by dpetka2001** [reply]
Yes, it will work under different sections, even in different files, as in the end all sections will get merged into the final spec of the plugin. For inspecting the settings of a plugin you can do `:lua vim.print(require("lazyvim.util").opts("nvim-cmp"))`. You can see there the value of `debounce`.
